core/src/main/java/org/apache/druid/common/config/NullValueHandlingConfig.java

```diff
@@ -32,7 +33,7 @@
   public NullValueHandlingConfig(@JsonProperty("useDefaultValueForNull") Boolean useDefaultValuesForNull)
   {
     this.useDefaultValuesForNull = useDefaultValuesForNull == null
-                                   ? true
+                                   ? Boolean.valueOf(System.getProperty(NULL_HANDLING_CONFIG_STRING, "true"))
```

Comment:
I don't think this is necessary. The normal Druid injectorator stuff (what is being activated by the `JsonConfigProvider.bind` line in NullHandlingModule) handles reading from system properties too.